src/main/java/decodex/Decodex.java

```diff
@@ -18,39 +31,62 @@
             + "| |_| | |_| |   <  __/\n"
             + "|____/ \\__,_|_|\\_\\___|\n";
 
+    /**
+     * Initializes the necessary Objects for Decodex.
+     */
+    public static void initDecodex() {
+        dataManager = new DataManager();
+        moduleManager = new ModuleManager();
+        parser = new Parser();
+        in = new Scanner(System.in);
+        ui = new Ui();
+    }
+
     /**
      * Decodex entry-point for the java.decodex.Decodex application.
      */
     public static void main(String[] args) {
-        DataManager dataManager = new DataManager();
-        ModuleManager moduleManager = new ModuleManager();
-        Ui ui = new Ui();
-
-        System.out.println("Hello from\n" + LOGO);
+        printGreeting();
+        initDecodex();
 
-        Scanner in = new Scanner(System.in);
-
-        // Temporary code, command functions will be moved to parser
         Command command = null;
 
         do {
-            System.out.print("Decodex > ");
+            printPromptHeader();
             String userInput = in.nextLine();
 
-            switch (userInput) {
-            case ExitCommand.COMMAND_WORD:
-                command = new ExitCommand(dataManager, moduleManager, ui);
-                break;
-            default:
-                // Skeletal - Just "echos" back to us.
-                System.out.println(userInput);
+            try {
+                command = parser.parseCommand(userInput, dataManager, moduleManager, ui);
+            } catch (ParserException err) {
+                printErrorMessage(err.getMessage());
                 continue;
             }
-
             command.run();
-
         } while (!(command instanceof ExitCommand));
 
         System.exit(0);
     }
```

Comment:
Follow up from the previous comment, perhaps move the main logic into a `run()` method? Then in the main method, we will simply do `new Decodex().run()`.